routes/index.js

```diff
@@ -35,6 +35,28 @@ const routes = [
     method: 'GET',
     handler: require('../handlers/search'),
   },
+
+  {
+    path: '/problems/{id?}',
```

Comment:
As the `id` is required, use `/problems/{id}` instead (same for the other routes)